fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreBase.java

```diff
@@ -1128,9 +1128,14 @@ default RecordCursor<FDBIndexedRecord<M>> scanIndexRecords(@Nonnull final Index
                 } catch (UnsupportedRemoteFetchIndexException ex) {
                     // In this case (e.g. the index maintainer does not support remote fetch), log as info
                     if (LOGGER.isInfoEnabled()) {
-                        LOGGER.info(KeyValueLogMessage.of("scanIndexRecords: Remote fetch unsupported, continuing with Index scan",
+                        KeyValueLogMessage msg = KeyValueLogMessage.build("scanIndexRecords: Remote fetch unsupported, continuing with Index scan",
                                 LogMessageKeys.MESSAGE, ex.getMessage(),
-                                LogMessageKeys.INDEX_NAME, index.getName()));
+                                LogMessageKeys.INDEX_NAME, index.getName());
+                        final SubspaceProvider subspaceProvider = getSubspaceProvider();
+                        if (subspaceProvider != null) {
```

Comment:
`getSubspaceProvider` is `@Nullable` on the `FDBRecordStoreBase` interface. However, none of the implementations return `null`, so it seems like something we could change. If we did, we could simplify this to avoid the null check.